protocol/tests/round_trips.rs

```diff
@@ -53,3 +58,100 @@ fn hello_world_happy_path() {
         .unwrap();
     assert_eq!(message, messages.message.unwrap());
 }
+
+#[test]
+#[cfg(feature = "std")]
+#[ignore = "CI"]
+fn regtest_handshake() {
```

Comment:
Wonder if we should put this in it's own file since it has a third party dependency?

Reply:
I would consider it an integration test even if it isn't ran in CI